### PR TITLE
Add code digest to autotask client

### PR DESCRIPTION
**Pull request by spalladino**
Add new codeDigest attribute to autotasks, and add a method for generating the digest of local code, so we can decide to skip an upload if it matches the current digest.

### CODE REVIEW

File: packages/autotask-client/src/zip.ts [hunk 1]
```diff
@@ -22,7 +22,9 @@ export async function zipFolder(folderPath: string): Promise<string> {
   const zip = new JSZip();
   for (const path of files) {
     const content = await promisify(readFile)(join(folderPath, path));
-    zip.file(path, content);
+    // We hardcode the date so we generate the same zip every time given the same contents
+    // This allows us to use the codedigest to decide whether or not to reupload code
+    zip.file(path, content, { date: new Date(2020, 1, 1, 0, 0, 0, 0) });
```
Comment (shahnami) [inline packages/autotask-client/src/zip.ts:27]: Curious why we need a date here, is codeDigest not only based on contents?

**Review by shahnami — APPROVED** [review]
LGTM, thanks!

File: packages/autotask-client/src/zip.ts (hunk 1 above)
Comment (spalladino) [reply packages/autotask-client/src/zip.ts:27]: Nope! The codeDigest in AWS Lambda is calculated from the zipfile, not from its contents. So the dates of the files within the zipfile actually affect the hash of the zip, and change the codeDigest. I can't explain how many hours I lost to that.

Comment (shahnami) [reply packages/autotask-client/src/zip.ts:27]: > Nope! The codeDigest in AWS Lambda is calculated from the zipfile, not from its contents. So the dates of the files within the zipfile actually affect the hash of the zip, and change the codeDigest. I can't explain how many hours I lost to that.

Jesus. Good to know! Thanks.

**Pull request merged**